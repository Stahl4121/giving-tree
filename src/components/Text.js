import React from 'react';
import Typography from '@material-ui/core/Typography';
import Box from '@material-ui/core/Box';
import { useEffect } from 'react';

export default function Text(props) {
  const [text, setText] = React.useState(props.text);
  const [type, setType] = React.useState(props.type);

  useEffect(() => { setText(props.text); }, [props.text]);
  useEffect(() => { setType(props.type); }, [props.type]);

  return (
    <div>
      {type === "card-heading" &&
        <Typography
          variant="h5"
        >
          {text}
        </Typography>
      }
      {type === "card-aboveheading" &&
        <Typography
          variant="button"
          color="primary" 
          component="p">
          {text}
        </Typography>
      }
      {type === "card-subheading" &&
        <Typography
          variant="body2"
          color="textSecondary" 
          component="p"
          gutterBottom>
          {text}
        </Typography>
      }
      {type === "tag" &&
        <Typography
          variant="body2">
          {text}
        </Typography>
      }
      {type === "progress-bold" &&
        <Typography
          variant="subtitle1">
          <Box fontWeight="fontWeightBold">
            {text}
          </Box>
        </Typography>
      }
      {type === "progress" &&
        <Typography
          variant="subtitle1">
          {text}
        </Typography>
      }
    </div>
  );
}

