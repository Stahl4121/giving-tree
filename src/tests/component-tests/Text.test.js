import React from "react";
import { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Text from "../../components/Text.js";

describe('Component: Text', () => {
  var enzyme = require('enzyme');
  enzyme.configure({ adapter: new Adapter() });

  it('Renders correctly', () => {
    const text = "my text";
    const type = 'h1';
    const wrap = enzyme.mount(<Text text={text} type={type}/>)
    expect(wrap).toMatchSnapshot();
  });

})
