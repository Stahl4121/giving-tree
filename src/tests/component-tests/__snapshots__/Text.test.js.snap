// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Text Renders correctly 1`] = `ReactWrapper {}`;
